src/main/java/pl/tomaszdziurko/jvm_bloggers/http/ProtocolSwitchingAwareConnectionRedirectHandler.java

```diff
@@ -97,7 +97,7 @@ private void setupConnection(HttpURLConnection conn, Map<String, List<String>> h
         conn.setReadTimeout(DEFAULT_TIMEOUT);
         // handle redirects within the same protocol
         conn.setInstanceFollowRedirects(true);
-        if (headers != null) {
+        if (headers != null && !headers.isEmpty()) {
```

Comment:
I think this change is not required at all. How it will improve existing code? If `headers` are empty then the loop in `setupHeaders` will just not execute, will it?

I think it's unnecessary in that case however in future you can use `[MapUtils.isEmpty(Map)](https://commons.apache.org/proper/commons-collections/javadocs/api-release/org/apache/commons/collections4/MapUtils.html#isEmpty(java.util.Map))` for maps or `[CollectionUtils.isEmpty(Collection)](https://commons.apache.org/proper/commons-collections/javadocs/api-release/org/apache/commons/collections4/CollectionUtils.html#isEmpty(java.util.Collection))` for general collection.